<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    

     <script>

          const promise = new Promise( (resolve, reject) => {
           
            setTimeout ( () => {
                
                    // reject("error occured");
                    resolve( { name:"abc",session:2024})
                 }, 3000)
           
           })          

        //   promise
        //      .then(
        //      (val)=> { console.log("value from response",val)},
        //      (error) =>{ console.log(error)}
        //        )
        //     // console.log(typeof val);
        //      .then(()=>console.log("2"))  //it will run only after resolve function is done

             promise
             .then(
             (val)=> { 
              console.log("value from response",val)
              // return new Promise((resolve,reject)=> resolve(val))
              //or
              return val;
            },
            
               )
             .then((val)=>console.log("2",val))  
             .catch(
             (error) =>{ console.log(error)}
             )
             .finally(()=>{console.log("promise is settled")})  //will always run
          
         function delayedAddition (num1, num2, delayTime) {

            return new Promise( (resolve,reject) => {
               
                setTimeout( ()=> {
                resolve(num1 + num2)}, delayTime*1000) //FOR MILLISECONDS SO SEC*1000


            })
         }

       delayedAddition(10,20,5)
       .then((result) => {console.log(result)})  //in result, value of resolve function is saved

// CALLBACK HELL

          
     </script>


</body>
</html>



